### PR TITLE
Species page sidebar data

### PR DESCRIPTION
**Pull request by msmanoj**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-753

## Description
This PR is to populate the species homepage sidebar by using sample data defined locally.

## Deployment URL
http://species-homepage-sidebar.review.ensembl.org/species/homo_sapiens_GCA_000001405_28

## Views affected
Species homepage -> Sidebar

### Other effects
- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

### CODE REVIEW

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 1]
```diff
@@ -41,3 +43,38 @@ export const human38stats = {
...
+      last_updated_date: 'Feb 2019',
+      gencode_version: 'GENCODE 34',
+      database_version: '100.38',
+      taxonomy_id: '9606'
+    },
+    psuedoautosomal_regions: {
+      description: `The Y chromosome in this assembly contains two psuedoautosomal regions (PARs) that were taken from the corresponding X chromosoles and are exact duplicates:
+        chrX:10,000-2,781,479 = chrY:10,000-2,781,479
+        chrX:155,701,382-156,030,895 = chrY:56,887,02-57,217,415 `
+    }
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:78]: As Andy said, the notice about pseudoautosomal regions is relevant only for a single species, which is human. I'm thinking that the responsibility for showing or hiding this notice may be hard-coded into the sidebar itself.

Unless we can generalise this block to `notes` or something, if some other species are also going to have some unique text blocks about them. We'll need to ask Andrea / Steve what they think.

File: src/ensembl/src/content/app/species/state/sidebar/speciesSidebarSelectors.ts [hunk 2]
```diff
@@ -14,7 +14,18 @@
...
 import { RootState } from 'src/store';
 
 export const isSidebarOpen = (state: RootState) =>
   state.speciesPage.sidebar.isOpen;
+
+export const getActiveGenomeSidebarPayload = (state: RootState) => {
+  const activeGenomeId = getActiveGenomeId(state);
+
+  if (!activeGenomeId) {
+    return;
```
Comment (azangru) [inline src/ensembl/src/content/app/species/state/sidebar/speciesSidebarSelectors.ts:27]: this makes it return `undefined`. Probably better return null for consistency with the next line?

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 3]
```diff
@@ -41,3 +43,38 @@ export const human38stats = {
...
+        name: 'INSDC Assembly',
+        id: 'GCA_000001405.28',
+        url: ''
+      },
+      level: 'complete genome'
+    },
+    annotation: {
+      provider: 'Ensembl',
+      method: 'Full genebuild',
+      last_updated_date: 'Feb 2019',
```
Comment (msmanoj) [inline src/ensembl/src/content/app/species/sample-data.ts:69]: TODO: Change to assembly_date

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 4]
```diff
@@ -41,3 +43,38 @@ export const human38stats = {
...
 };
+
+type SpeciesSidebarData = {
+  [genomeId: string]: SpeciesSidebarPayload;
+};
+
+export const sidebarData: SpeciesSidebarData = {
+  homo_sapiens_GCA_000001405_28: {
+    species: {
+      display_name: 'Human',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:54]: common_name

File: src/ensembl/src/content/app/species/state/sidebar/speciesSidebarSelectors.ts (hunk 2 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/state/sidebar/speciesSidebarSelectors.ts]: That's done

**Comment by azangru** [comment]
I pushed a commit with the following changes:
- removed margin-bottom from ExternalReference (we can easily set it where needed by passing classname from parent)
- removed unused CSS classes (some were left from when you split ExternalReference into ExternalReference + ExternalLink)
- updated names of classes — removed the word `default`
- applied date formatting using the browser Intl api to let the browser transform the string `2019-02-01` to 'Feb 2019'

We'll still need to check how the ExternalReference component behaves in other parts of our sites.

**However!** I am pretty sure we do not want to add a link to the annotation provider if it is Ensembl — that would be just linking to ourselves. We may want to link to providers that are not Ensembl though. Should clarify this with Andrea.

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 4 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    label: string;
+    value?: string | null;
+    url?: string;
+  };
+  const assemblyDate = new Date(payload.assembly_date);
+  const formattedAssemblyDate = Intl.DateTimeFormat('en-GB', {
+    month: 'short',
+    year: 'numeric'
+  }).format(assemblyDate);
+  const annotationEntriesMap: AnnotationEntry[][] = [
```
Comment (azangru) [inline src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:62]: It's not even a map 😕 . `annotationEntriesMap` -> `annotationEntries`?

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 6]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+    scientific_name: 'Homo sapiens',
+    gencode_version: 'GENCODE 34',
+    assembly_name: 'GRCh38.p13',
+    assembly_provider: {
+      name: 'INSDC assembly',
+      url: ''
+    },
+    annotation_provider: {
+      name: 'Ensembl',
+      url: 'www.ensembl.org'
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:564]: Probably want to add `https://`; without the `//` part the link behaves as a relative link.

Don't know what should happen when the urls start arriving via an api. Ideally, the backend should also know the protocol for reaching this resource; we shouldn't be guessing it on the frontend. Alternatively, we can use `http` everywhere, and let the target resources redirect to `https` if they so wish.

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+              {entries.map((entry, entry_index) => {
+                const labelClassNames = classNames(styles.label, {
+                  [styles.labelShort]: group_index === 0
+                });
+
+                return entry.value ? (
+                  <div key={entry_index} className={styles.standardLabelValue}>
+                    <div className={labelClassNames}>{entry.label}</div>
+                    <div className={styles.value}>
+                      {entry.url ? (
```
Comment (azangru) [inline src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:128]: We should either skip external reference if the provider is Ensembl, or link to some kind of about page.

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 8]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+    scientific_name: null,
+    gencode_version: null,
+    assembly_name: 'WBcel235',
+    assembly_provider: {
+      name: 'INSDC assembly',
+      url: ''
+    },
+    annotation_provider: {
+      name: 'WormBase',
+      url: 'www.ensembl.org'
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:639]: http://www.wormbase.org

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 9]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+    common_name: 'Plasmodium falciparum 3D7',
+    scientific_name: null,
+    gencode_version: null,
+    assembly_name: 'ASM276v2',
+    assembly_provider: {
+      name: 'INSDC assembly',
+      url: ''
+    },
+    annotation_provider: {
+      name: 'Naval Medical Research Institute',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:660]: In the db, this is assembly provider, not annotation provider. Curiously, there is no record of an annotation provider for Plasmodium falciparum:

image

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 10]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+    scientific_name: 'Saccharomyces cerevisiae',
+    gencode_version: null,
+    assembly_name: 'R64',
+    assembly_provider: {
+      name: 'INSDC assembly',
+      url: ''
+    },
+    annotation_provider: {
+      name: 'Saccharomyces Genome Database',
+      url: 'www.ensembl.org'
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:686]: http://www.yeastgenome.org

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 11]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+      type: 'Strain',
+      value: 'S288C'
+    },
+    database_version: '101.4',
+    common_name: 'Brewers yeast',
+    scientific_name: 'Saccharomyces cerevisiae',
+    gencode_version: null,
+    assembly_name: 'R64',
+    assembly_provider: {
+      name: 'INSDC assembly',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:681]: According to the database, assembly provider is also Saccharomyces Genome Database

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 12]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+    scientific_name: 'Triticum aestivum',
+    gencode_version: null,
+    assembly_name: 'IWGSC',
+    assembly_provider: {
+      name: 'INSDC assembly',
+      url: ''
+    },
+    annotation_provider: {
+      name: 'International Wheat Genome Sequencing Consortium',
+      url: 'www.ensembl.org'
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:711]: https://www.ebi.ac.uk/ena/data/view/GCA_900519105.1

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 13]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+      type: 'Cultivar',
+      value: 'Chinese spring'
+    },
+    database_version: '101.4',
+    common_name: 'Wheat',
+    scientific_name: 'Triticum aestivum',
+    gencode_version: null,
+    assembly_name: 'IWGSC',
+    assembly_provider: {
+      name: 'INSDC assembly',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:706]: According to the DB, assembly provider for wheat is the same as annotation provider

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 14]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+      type: 'Strain',
+      value: 'N2'
+    },
+    database_version: '101.269',
+    common_name: 'Caenorhabditis elegans',
+    scientific_name: null,
+    gencode_version: null,
+    assembly_name: 'WBcel235',
+    assembly_provider: {
+      name: 'INSDC assembly',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:634]: It is WormBase, same as annotation provider

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 15]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+    annotation_method: 'full genebuild',
+    assembly_date: 'June 2013',
+    notes: []
+  },
+  escherichia_coli_str_k_12_substr_mg1655_GCA_000005845_2: {
+    id: 'GCA_000005845.2',
+    taxonomy_id: '511145',
+    strain: null,
+    database_version: '101.1',
+    common_name: 'Escherichia coli str. K-12 substr. MG1655',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:604]: Umm, this is scientific name; common_name should be null.

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 16]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+  },
+  caenorhabditis_elegans_GCA_000002985_3: {
+    id: 'GCA_000002985.3',
+    taxonomy_id: '6239',
+    strain: {
+      type: 'Strain',
+      value: 'N2'
+    },
+    database_version: '101.269',
+    common_name: 'Caenorhabditis elegans',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:629]: This is scientific name; common_name should be null.

File: src/ensembl/src/content/app/species/sample-data.ts [hunk 17]
```diff
@@ -538,3 +540,179 @@ export const sampleData: RawSpeciesStats = {
...
+    annotation_method: 'Import',
+    assembly_date: 'Feb 2013',
+    notes: []
+  },
+  plasmodium_falciparum_GCA_000002765_2: {
+    id: 'GCA_000002765.2',
+    taxonomy_id: '36329',
+    strain: null,
+    database_version: '101.1',
+    common_name: 'Plasmodium falciparum 3D7',
```
Comment (azangru) [inline src/ensembl/src/content/app/species/sample-data.ts:651]: This is scientific name; common_name should be null.

Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 16 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 15 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 14 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 13 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 12 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 11 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 10 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 9 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 8 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That's done

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx (hunk 5 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx]: That's done

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 15 above)
Comment (azangru) [reply src/ensembl/src/content/app/species/sample-data.ts]: Right. There's immediately a problem in the interface 😉 

image

Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: Ah yes! Please wait until tomorrow before checking the UI :D

**Comment by azangru** [comment]
I've pushed a commit to fix ExternalReferences in Protein view.

Before:

image



After:
image

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 15 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: That should now be fixed

File: src/ensembl/src/content/app/species/sample-data.ts (hunk 6 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/sample-data.ts]: I've added `http://` to the missing ones

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx (hunk 7 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx]: I've removed the link and just displaying Ensembl.

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+  ];
+
+  return (
+    <div>
+      <div className={styles.speciesDetails}>
+        {payload.common_name && (
+          <span className={styles.common_name}>{payload.common_name}</span>
+        )}
+        {payload.scientific_name && (
+          <span className={styles.scientific_name}>
```
Comment (azangru) [inline src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:88]: Neither the common_name (why not commonName?) class, nor the scientific_name class exists in the styles file.

I would actually argue that targeting these spans using class names is better then using their positions relative to one another.

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+
+      {payload.strain && (
+        <div className={styles.strainDetails}>
+          <span className={styles.strain_type}>{payload.strain.type}</span>
+          <span className={styles.strain_value}>{payload.strain.value}</span>
+        </div>
+      )}
+
+      <div className={styles.sectionHead}>Assembly</div>
+      <div className={styles.assemblyDetails}>
```
Comment (azangru) [inline src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:102]: bit of a strange spacing here:



compare it with the design:

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx (hunk 18 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx]: That's done

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx (hunk 19 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:102]: Updated the styles to match the design

File: src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx [hunk 20]
```diff
@@ -0,0 +1,175 @@
...
+  ];
+
+  return (
+    <div>
+      <div className={styles.speciesDetails}>
+        {payload.common_name && (
+          <span className={styles.commonName}>{payload.common_name}</span>
+        )}
+        {payload.scientific_name && (
+          <span className={styles.scientificName}>
```
Comment (azangru) [inline src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:88]: We are back to our old friend 😃 :

image

I think we need a function/component that's going to be called displayName, and that's put common name or scientific name into spans with classes namePrimary/nameSecondary.

Comment (msmanoj) [reply src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:88]: Would this be simpler instead?
https://github.com/Ensembl/ensembl-client/pull/364/commits/185e462d9246a5712ab742065be3b198292a38a5

Comment (azangru) [reply src/ensembl/src/content/app/species/components/species-sidebar/SpeciesSidebar.tsx:88]: Yep, even better!

**Review by azangru — APPROVED** [review]

**Pull request merged**